004-testing.md

```diff
@@ -0,0 +1,50 @@
+# Testing
+
+We write automated tests to help us catch mistakes before they impact our users. Automated tests also allow us to test more thoroughly and quickly than if we were testing "by hand", which lets us iterate faster & more confidently!
+
+**Discussion:** https://github.com/DoSomething/rfcs/pull/4
+
+## Problem
+
+We've found that when we skip writing tests, we're less confident in changes we make. That eventually slows down feature development and discourages refactoring. On the other hand, we don't want to spend all of our time writing tests, since we recognize that they're just a tool (and don't directly provide value to our users).
+
+## Proposal
+
+Well, we should write tests, of course! Here's our current "best practices" for testing:
+
+### Philosophy
+There are lots of opinions on how to write tests! We try be pragmatic.
+We write tests to save ourselves from doing tedious QA work, and our tests reflect that: they're the steps that an engineer or PM would use to test a given function, component, API endpoint, or user flow... but translated into a set of instructions so a machine can do it for us!
+
+We [don't practice strict TDD](https://dhh.dk/2014/tdd-is-dead-long-live-testing.html) or pursue 100% test coverage. Instead, we try to get the most "bang for our buck" by writing [fewer higher-level integration tests](https://youtu.be/Fha2bVoC8SE) that test the real functionality that our application exposes, rather than isolated pieces.
```

Comment:
I definitely think it's difficult to find the right balance for writing tests, but setting some kind of concrete goal would be helpful for us to measure our success. Or even just to see whether 'x' amount of testing is impactful, etc.

Reply:
Yeah, that's totally reasonable. My concern with making test coverage into a tracked metric is that it'd be easy for that to then become a goal in and of itself (when it's really just a tool for helping us to work more efficiently).

At least for now, I like the plan we settled on in our testing sync: to be more thoughtful about making time for writing tests when estimating tickets & to let the "did you add tests" checklist on PRs shame us if we don't. 😼